### PR TITLE
Stop browse thread before deleting the library

### PR DESCRIPTION
**Pull request by daschuer**
This will fix https://github.com/mixxxdj/mixxx/issues/11589

### CODE REVIEW

**Comment by ronso0** [comment]
IIUC this isn't stopping the BrowseThread, it just resets the BrowseThreadPointer to prevent further calls _from_ and callbacks _to_ the BrowseFeature.

Is real stopping required? (didn't understand the actual bug tbh)
If yes, that could be achieved via `m_bStopThread` like in the BrowseThread destructor.
If no, the new method should probably be renamed.

**Comment by daschuer** [comment]
Yes, the code is not obvious. I have added a comment. 

The issue was that if the browse thread is using the library and when you quit Mixxx in the mean time from the main thread. It is using dangling pointers and crashes Mixxx. 
The fix is to stop these threads and wait until they are really terminated and then continue destructing the library. 
This is done here implicit by releasing all references to the browse thread (The last reset() calls immediately the destructor). 

Currently the recording feature and the browse feature are using this thread. The recording feature is destructed earlier because it is places in the widget tree of the skin. This asymmetry is not obvious and bites me during development, but a fix would involve a bigger refactoring. Probably required when moving to QML, but not in the stable 2.3 branch.

**Review by ronso0 — APPROVED** [review]
LGTM. Thanks for the QSharedPointer explanation.

File: src/library/browse/browsetablemodel.cpp [hunk 1]
```diff
@@ -471,3 +473,10 @@ QAbstractItemDelegate* BrowseTableModel::delegateForColumn(const int i, QObject*
     }
     return nullptr;
 }
+
+void BrowseTableModel::stopBrowseThread() {
+    // The shared browse thread is actually stopped in the desctuctor
```
Comment (uklotzde) [inline src/library/browse/browsetablemodel.cpp:478]: Typo

File: src/library/browse/browsetablemodel.cpp [hunk 2]
```diff
@@ -471,3 +473,10 @@ QAbstractItemDelegate* BrowseTableModel::delegateForColumn(const int i, QObject*
     }
     return nullptr;
 }
+
+void BrowseTableModel::stopBrowseThread() {
```
Comment (uklotzde) [inline src/library/browse/browsetablemodel.cpp:477]: The name of this function is inappropriate as the comment tells. It doesn't stop the browse thread, it just drops one of multiple references.

-> `resetBrowseThread()` or `releaseBrowseThread()`?

**Comment by daschuer** [comment]
Done.

**Comment by ronso0** [comment]
Thank you!

**Pull request merged**